scripts/main/dragDrop.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * drag album to another one
+ * @param {DragEvent} ev
+ */
+function startAlbumDrag(ev) {
+	ev.dataTransfer.setData("text", ev.target.id);
+}
+
+/**
+ * drop album
+ * @param {DragEvent} ev
+ */
+function dropAlbum(ev) {
+	ev.preventDefault();
+	if (album.isSmartID(ev.target.id.substring(6))) return;
```

Comment:
Yeah, that's what I meant in an earlier comment about not providing `ondrop` for smart and tag albums in the first place -- I think that way you wouldn't need this test here at all?

As to your more general question/comment about it working weird for the corners or smart albums, it's not just the corners and not just the smart albums. Whatever the reason might be, `ev.target` is not necessarily `<div>` (or even `<img>`), but if you drop, say, right over the album title, it will be `<h1>`. Adding `id` to all of them would be insane (which is part of the reason why I objected to it in the first place). But there should be an easy fix, something like:
```
    albumdiv = ev.target.closest("div.album");
```
and then you can again use `dataset.id` and `className`.

Also, you should check if the source and the destination are not the same album.